app/src/main/java/campus/tech/kakao/map/ui/activity/SearchActivity.kt

```diff
@@ -0,0 +1,103 @@
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class SearchActivity : AppCompatActivity() {
+    private lateinit var binding: ActivitySearchBinding
+    private lateinit var placesAdapter: PlacesAdapter
+    private lateinit var searchHistoryAdapter: SearchHistoryAdapter
+
+    private val viewModel: MapViewModel by viewModels ()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySearchBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        setUpSearchHistoryAdapter()
+        setUpPlacesAdapter()
+        setUpViewModelObservers()
+
+        binding.searchInput.addTextChangedListener { text ->
+            viewModel.searchPlaces(text.toString())
+//            viewModel.searchDBPlaces(text.toString())     // DBHelper 테스트 용 (검색)
+//            viewModel.searchRoomPlaces(text.toString())   // Room 테스트 용 (검색)
+        }
+
+        binding.deleteInput.setOnClickListener {
+            binding.searchInput.text.clear()
+        }
+    }
+
+    private fun setUpSearchHistoryAdapter() {
+        searchHistoryAdapter = SearchHistoryAdapter(
+            onDeleteClick = { position: Int ->
+                viewModel.delSearch(position)
+            },
+            onTextClick = { position: Int ->
+                val itemName = searchHistoryAdapter.getItemName(position)
+                binding.searchInput.setText(itemName)
+            })
+        binding.searchHistory.adapter = searchHistoryAdapter
+    }
+
+    private fun setUpPlacesAdapter() {
+        placesAdapter = PlacesAdapter { item: Place ->
+            val itemName = item.name
+            lifecycleScope.launch {
+                viewModel.insertSearch(itemName)
+                viewModel.savePos(item.longitude, item.latitude)
+                Log.d("prefs", "lifecycle: ${Thread.currentThread().name}")
+            }
+            Log.d("prefs", "lifecycle2: ${Thread.currentThread().name}")
+            goToSearch(item)
+        }
+        binding.placesRView.adapter = placesAdapter
+        binding.placesRView.layoutManager = LinearLayoutManager(this)
+    }
+
+    private fun setUpViewModelObservers() {
+        viewModel.places.observe(this, Observer { places ->
+            placesAdapter.updateList(places)
+            placesAdapter.notifyDataSetChanged()
+            binding.textView.visibility =
+                if (placesAdapter.itemCount <= 0) View.VISIBLE else View.GONE
+        })
+        
+        viewModel.searchHistoryData.observe(this, Observer { searchHistoryData ->
+            searchHistoryAdapter.submitList(searchHistoryData.toList())
+            binding.searchHistory.isVisible = searchHistoryData.isNotEmpty()
+        })
+    }
+
+    private fun goToSearch(place: Place) {
+        val intent = Intent(this, MainActivity::class.java).apply {
+            putExtra("place", place)
+        }
```

Comment:
이 때의 Intent는 데이터를 담는 역할만을 수행하기 때문에 아래와 같이 파라미터가 없는 생성자를 호출해 Intent를 생성하는 것이 좋습니다.
```suggestion
        val intent = Intent().apply {
            putExtra("place", place)
        }
```